checkov/cdk/checks/typescript/LambdaInVPC.yaml

```diff
@@ -0,0 +1,51 @@
+metadata:
+  version: 0.2
+  approach: define failing
+  id: CKV_AWS_117
+  name: Ensure that AWS Lambda function is configured inside a VPC
+  category: GENERAL_SECURITY
+  framework: cdk
+scope:
+  languages:
+    - typescript
+definition:
+  or:
+    - pattern: new $PACKAGE.aws_lambda.Function(<ANY>, <ANY>, { <ANY> })
+      conditions:
+        - not_pattern: |
+            new $PACKAGE.aws_lambda.Function(<ANY>, <ANY>, { <ANY>, vpc: <ANY>, <ANY> } )
+        - metavariable: $PACKAGE
+          comparison: $PACKAGE == "aws-cdk-lib"
+    - pattern: new $PACKAGE.aws_lambda.CfnFunction(<ANY>, <ANY>, { <ANY> })
+      conditions:
+        - not_pattern: |
+            new $PACKAGE.aws_lambda.CfnFunction(<ANY>, <ANY>, { <ANY>, vpcConfig: <ANY>, <ANY> } )
+        - metavariable: $PACKAGE
+          comparison: $PACKAGE == "aws-cdk-lib"
+    - pattern: new $PACKAGE.aws_sam.CfnFunction(<ANY>, <ANY>, { <ANY> })
+      conditions:
+        - not_pattern: |
+            new $PACKAGE.aws_sam.CfnFunction(<ANY>, <ANY>, { <ANY>, vpcConfig: <ANY>, <ANY> } )
+        - metavariable: $PACKAGE
+          comparison: $PACKAGE == "aws-cdk-lib"
+    - pattern: |
```

Comment:
not sure what's the order inside the engine to find patterns and conditions but it seems too generic to me, it will catch almost any object declaration and check the condition on it

Reply:
yes, that why i give hime more strict pattern `aws_sam.CfnFunction`, it's hard to catch it if you are not working in the same package. 
you should see the python cdk and decide how is better